### PR TITLE
update RFC template to make required sections explicit

### PR DESCRIPTION
**Pull request by brijk7**

### CODE REVIEW

File: rfcs/yyyymmdd-rfc-template.md [hunk 1]
```diff
@@ -34,15 +34,40 @@ idea, and list pros/cons to each approach. If there are alternatives that you
...
+* Dependencies: does this proposal add any new dependencies to TensorFlow?
+* Dependent projects: are there other areas of TensorFlow or things that use TensorFlow (TFX/pipelines, TensorBoard, etc.) that this affects? How have you identified these dependencies and are you sure they are complete? If there are dependencies, how are you managing those changes?
+
+### Engineering Impact
+* Do you expect changes to binary size / startup time / build time / test times?
+* Who will maintain this code? Is this code in its own buildable unit? Can this code be tested in its own? Is visibility suitably restricted to only a small API surface for others to use?
+
+### Platforms and Environments
+* Platforms: does this work on all platforms supported by TensorFlow? If not, why is that ok? Will it work on embedded/mobile? Does it impact automatic code generation or mobile stripping tooling? Will it work with transformation tools?
+* Execution environments (CloudML, Cloud TPUs): what impact do you expect and how will you confirm?
```
Comment (ewilderj) [inline rfcs/yyyymmdd-rfc-template.md]: Could we expand the list of examples here to not just be Google Cloud products? Or to be more generic, eg. "Cloud services, accelerator hardware".

File: rfcs/yyyymmdd-rfc-template.md [hunk 2]
```diff
@@ -34,15 +34,40 @@ idea, and list pros/cons to each approach. If there are alternatives that you
...
-Factors to consider include:
-
-* performance implications
-* dependencies
-* maintenance
-* platforms and environments impacted (e.g. hardware, cloud, other software
-  ecosystems)
-* [compatibility](https://www.tensorflow.org/programmers_guide/version_compat)
-* how will this change impact users, and how will that be managed?
+Make sure you’ve thought through and addressed the following sections (if a section is not relevant to your specific proposal, please explain why).
```
Comment (ewilderj) [inline rfcs/yyyymmdd-rfc-template.md]:
```suggestion
Make sure you’ve thought through and addressed the following sections. If a section is not relevant to your specific proposal, please explain why, e.g. your RFC addresses a convention or process, not an API.
```

adding this to accommodate that people will still write RFCs for things that aren't just API design, e.g. processes

File: rfcs/yyyymmdd-rfc-template.md (hunk 1 above)
Comment (brijk7) [reply rfcs/yyyymmdd-rfc-template.md]: Done.

File: rfcs/yyyymmdd-rfc-template.md (hunk 2 above)
Comment (brijk7) [reply rfcs/yyyymmdd-rfc-template.md]: Done.

**Pull request merged**